packages/server/src/websocket/services/MessagePersister.ts

```diff
@@ -1,62 +1,143 @@
+      const result = await (this.redis.set as any)(this.lockKey, this.instanceId, 'NX', 'PX', this.lockTTL)
+
+      // If result is OK, lock was acquired successfully
+      if (result === 'OK') {
+        this.logger.debug(`[acquireLock] Lock acquired successfully by instance ${this.instanceId}`)
+        return true
+      }
+
+      // Otherwise, log the current lock holder
+      const currentLockHolder = await this.redis.get(this.lockKey)
+      this.logger.debug(`[acquireLock] Lock is currently held by instance: ${currentLockHolder}`)
+      return false
+    } catch (error) {
+      this.logger.error(`[acquireLock] Error acquiring lock: ${error}`)
+      return false
+    }
+  }
+
+  // Periodically renews the lock to maintain mastership
+  renewLockPeriodically() {
+    this.renewInterval = setInterval(async () => {
+      try {
+        // Check if this instance still holds the lock before renewing
+        const currentLockHolder = await this.redis.get(this.lockKey)
+        if (currentLockHolder === this.instanceId) {
+          await this.redis.pexpire(this.lockKey, this.lockTTL)
+          this.logger.debug(`[renewLockPeriodically] Lock renewed by instance ${this.instanceId}`)
+        } else {
+          this.logger.warn(`[renewLockPeriodically] Lock is no longer held by instance ${this.instanceId}`)
+          this.clearMasterShip() // Relinquish mastership if another instance took the lock
+        }
+      } catch (error) {
+        this.logger.error(`[renewLockPeriodically] Error renewing lock: ${error}`)
+        this.clearMasterShip() // Relinquish mastership if unable to renew
+      }
+    }, this.lockTTL / 2) // Renew before the TTL expires
+  }
+
+  // Clear mastership if lock renewal fails or another instance acquires it
+  clearMasterShip() {
+    if (this.renewInterval) {
+      clearInterval(this.renewInterval)
+      this.renewInterval = null
+    }
+    if (!this.checkInterval) {
+      this.initiateMasterRole() // Restart attempting to acquire mastership if lost
+    }
+    this.logger.log('[clearMasterShip] Mastership relinquished.')
+  }
+
+  // Releases the lock when the module is destroyed, allowing another instance to become master
+  onModuleDestroy() {
+    if (this.renewInterval) {
+      clearInterval(this.renewInterval)
+    }
+    if (this.checkInterval) {
+      clearInterval(this.checkInterval)
+    }
+    this.redis.del(this.lockKey) // Releases the lock on shutdown
+    this.logger.log(`[onModuleDestroy] Lock released by instance ${this.instanceId}`)
+  }
+
+  // Starts the monitoring and migration process if this instance is the master
   startMonitoring() {
     this.logger.log(`[startMonitoring] Initialize MessagePersister`)
     this.thresholdTimestamp = this.configService.get<number>('appConfig.thresholdTimestamp', 60000)
     setInterval(() => this.migrateData(), this.thresholdTimestamp)
   }
 
+  // Migrates messages from Redis to MongoDB if they meet the age threshold
```

Comment:
In my opinion, this method should be called `persistMessages` or something similar to clarify its purpose, since even the term is 'data migration' is correct, I find it clearer to make explicit the purpose of this migration.

Reply:
Resolved! The method name is changed from migrateData to persistMessages.